test/util/MessagesSpec.scala

```diff
@@ -0,0 +1,187 @@
+      same.size.toDouble / defaultMessages.size.toDouble < 0.06 shouldBe true
+    }
+    "have a non-empty message for each key" in {
+      assertNonEmptyValuesForDefaultMessages()
+      assertNonEmptyValuesForWelshMessages()
+    }
+    "have no unescaped single quotes in value" in {
+      assertCorrectUseOfQuotesForDefaultMessages()
+      assertCorrectUseOfQuotesForWelshMessages()
+    }
+    "have a resolvable message for keys which take args" in {
+      val englishWithArgsMsgKeys = defaultMessages collect { case (key, value) if countArgs(value) > 0 => key }
+      val welshWithArgsMsgKeys = welshMessages collect { case (key, value) if countArgs(value) > 0     => key }
+      val missingFromEnglish = englishWithArgsMsgKeys.toList diff welshWithArgsMsgKeys.toList
+      val missingFromWelsh = welshWithArgsMsgKeys.toList diff englishWithArgsMsgKeys.toList
+      missingFromEnglish foreach { key =>
+        println(s"Key which has arguments in English but not in Welsh: $key")
+      }
+      missingFromWelsh foreach { key =>
+        println(s"Key which has arguments in Welsh but not in English: $key")
+      }
+      englishWithArgsMsgKeys.size shouldBe welshWithArgsMsgKeys.size
+    }
+    "have the same args in the same order for all keys which take args" in {
+      val englishWithArgsMsgKeysAndArgList = defaultMessages collect {
+        case (key, value) if countArgs(value) > 0 => (key, listArgs(value))
+      }
+      val welshWithArgsMsgKeysAndArgList = welshMessages collect {
+        case (key, value) if countArgs(value) > 0 => (key, listArgs(value))
+      }
+      val mismatchedArgSequences = englishWithArgsMsgKeysAndArgList collect {
+        case (key, engArgSeq) if engArgSeq != welshWithArgsMsgKeysAndArgList(key) =>
+          (key, engArgSeq, welshWithArgsMsgKeysAndArgList(key))
+      }
+      mismatchedArgSequences foreach {
+        case (key, engArgSeq, welshArgSeq) =>
+          println(
+            s"key which has different arguments or order of arguments between English and Welsh: $key -- English arg seq=$engArgSeq and Welsh arg seq=$welshArgSeq")
+      }
+      mismatchedArgSequences.size shouldBe 0
+    }
+  }
+
+  private def isInteger(s: String): Boolean = s forall Character.isDigit
+
+  private def toArgArray(msg: String) = msg.split("\\{|\\}").map(_.trim()).filter(isInteger(_))
+
+  private def countArgs(msg: String) = toArgArray(msg).size
+
+  private def listArgs(msg: String) = toArgArray(msg).mkString
+
+  private def assertNonEmptyValuesForDefaultMessages() = assertNonEmptyNonTemporaryValues("Default", defaultMessages)
+
+  private def assertNonEmptyValuesForWelshMessages() = assertNonEmptyNonTemporaryValues("Welsh", welshMessages)
+
+  private def assertCorrectUseOfQuotesForDefaultMessages() = assertCorrectUseOfQuotes("Default", defaultMessages)
+
+  private def assertCorrectUseOfQuotesForWelshMessages() = assertCorrectUseOfQuotes("Welsh", welshMessages)
+
```

Comment:
These 4 assert methods above can be inlined as they are only used in one place, might make the tests a bit easier to read